app/assets/javascripts/webchat.js

```diff
@@ -0,0 +1,112 @@
+(function () {
+  'use strict';
+  var root = this;
+  var $ = root.jQuery;
+  var windowLocationPathname = root.location.pathname;
+  var windowOpen = root.open;
+  if (typeof root.GOVUK === 'undefined') { root.GOVUK = {}; }
+  var GOVUK = root.GOVUK;
+
+  // Each page will have a different entryPointID, which is a queue id.
+  // Don't enable this plugin on pages that don't exist in this map, and
+  // uncomment the additional routes as we obtain them.
+  // Whether the actual template is displayed is in `contact_presenter#show_webchat?`.
+  var entryPointIDs = {};
+  // entryPointIDs['/government/organisations/hm-revenue-customs/contact/national-insurance-numbers'] =;
+  // entryPointIDs['/government/organisations/hm-revenue-customs/contact/self-assessment-online-services-helpdesk'] =;
+  // entryPointIDs['/government/organisations/hm-revenue-customs/contact/self-assessment'] =;
+  entryPointIDs['/government/organisations/hm-revenue-customs/contact/tax-credits-enquiries'] = 1012;
+  // entryPointIDs['/government/organisations/hm-revenue-customs/contact/vat-enquiries'] =;
+
+  var API_URL = 'https://online.hmrc.gov.uk/webchatprod/egain/chat/entrypoint/checkEligibility/';
+  var OPEN_CHAT_URL = function (entryPointID) {
+    return 'https://online.hmrc.gov.uk/webchatprod/templates/chat/hmrc6/chat.html?entryPointId=' + entryPointID + '&templateName=hmrc6&languageCode=en&countryCode=US&ver=v11';
+  };
+  var CODE_AGENTS_AVAILABLE = 0;
+  var CODE_AGENTS_UNAVAILABLE = 1;
+  var CODE_AGENTS_BUSY = 2;
+  var POLL_INTERVAL = 15 * 1000;
+
+  function Webchat (options) {
+    var $el = $(options.$el);
+    var $advisersUnavailable = $el.find('.js-webchat-advisers-unavailable');
+    var $advisersBusy = $el.find('.js-webchat-advisers-busy');
+    var $advisersAvailable = $el.find('.js-webchat-advisers-available');
+    var $openButton = $el.find('.js-webchat-open-button');
+
+    var entryPointID = entryPointIDs[windowLocationPathname];
+    var pollingEnabled = true;
+
+    if (entryPointID) {
+      pollAvailability();
+
+      $openButton.on('click', handleOpenChat);
+    }
+
+    function pollAvailability () {
+      checkAvailability();
+
+      setTimeout(function () {
```

Comment:
If the checkAvailability endpoint takes more than 15s to respond, this could stack up checks?

Should we have a callback in checkAvailability, and/or a timeout on the ajax call?

(Don't think the risk is so huge here though)